plugins/clickhouse/include/transformers.hpp

```diff
@@ -0,0 +1,151 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "clickhouse/columns/column.h"
+#include "tenzir/detail/enumerate.hpp"
+#include "tenzir/detail/heterogeneous_string_hash.hpp"
+#include "tenzir/detail/stable_map.hpp"
+#include "tenzir/diagnostics.hpp"
+#include "tenzir/generator.hpp"
+#include "tenzir/table_slice.hpp"
+
+// TODO: This should go into table_slice.hpp or somewhere like that
+namespace tenzir {
+struct column_view {
+  std::string_view name;
+  const tenzir::type& type;
+  const arrow::Array& array;
+};
+
+inline auto columns_of(const table_slice& slice) -> generator<column_view> {
+  const auto& schema = as<record_type>(slice.schema());
+  for (auto [i, kt] : detail::enumerate(schema.fields())) {
+    const auto& [k, _] = kt;
+    auto offset = tenzir::offset{};
+    offset.push_back(i);
+    auto [t, arr] = offset.get(slice);
+    co_yield column_view{k, t, *arr};
+  }
```

Comment:
Why so complicated instead of just this?

```cpp
const auto& rt = as<record_type>(slice.schema());
const auto& batch = to_record_batch(slice);
for (auto i = 0; i < rt.num_fields(); ++i) {
  auto field = rt.field(i);
  co_yield column_view{
    .name = field.name,
    .type = field.type,
    .array = *batch->GetColumn(i),
  };
}
```